File: 58-length-of-last-word/length-of-last-word.dart
class Solution {
  int lengthOfLastWord(String s) {
        int p = s.length -1;
        
        // Son Elemani Bul
        while (s[p] == " "){
            p--;  
        }
        int lastCharOfLastWord = p;

        // Ilk bosluga rastlayana kadar sola ilerle
        while (p >=0 && s[p] != " "){
            p--;
        }
        int firstCharOfLastWord = p;


      return lastCharOfLastWord - firstCharOfLastWord;
  }
}

File: 58-length-of-last-word/length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        p = len(s)-1
        e = 0

        while s[p].isspace():
            p -= 1
        
        e = p

        while p > 0 and s[p-1].isalpha():
            p -= 1
        
        return e - p + 1


        